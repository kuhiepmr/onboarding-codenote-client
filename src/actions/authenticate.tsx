export const types = {
  AUTHENTICATE: 'AUTHENTICATE',
};

export const authenticate = (payload) => ({
  type: types.AUTHENTICATE,
  payload
})