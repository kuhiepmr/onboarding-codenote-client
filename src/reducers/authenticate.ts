import { actionTypes } from "../actions/authenticate";

interface IStates {
  isAuthenticated: boolean;
}

export interface IActions {
  type: string;
  payload: Object;
}

const initialState = {
  isAuthenticated: false
};

export default function authenticate(
  state: IStates = initialState,
  action: IActions
) {
  switch (action.type) {
    case actionTypes.LOGIN_SUCCESS:
      return {
        ...state,
        isAuthenticated: action.payload
      };
    default:
      return state;
  }
}
