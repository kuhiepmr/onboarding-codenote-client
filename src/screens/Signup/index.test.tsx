import { shallow } from 'enzyme';
import { History } from 'history';
import React from 'react';
import { userHasAuthenticated } from '../../actions/authenticate';
import SignUp from './index';

describe('SignUp screen', () => {
  let wrapper, instance
  const props = {
    history: {} as History,
    userHasAuthenticated: userHasAuthenticated
  }

  beforeEach(() => {
    wrapper = shallow(<SignUp {...props} />);
    instance = wrapper.instance();
  });

  it('should render success', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should change field and validate', () => {
    const emailEvent = {
      target: {
        id: 'email',
        value: "admin@example.com"
      }
    }
    const passwordEvent = {
      target: {
        id: 'password',
        value: "123456"
      }
    }
    const confirmPasswordEvent = {
      target: {
        id: 'confirmPassword',
        value: "123456"
      }
    }

    instance.handleChange(emailEvent);
    instance.handleChange(passwordEvent);
    instance.handleChange(confirmPasswordEvent);

    expect(wrapper.state().newUser).toBe(null);
    expect(wrapper.state().email).toEqual('admin@example.com');
    expect(wrapper.state().password).toEqual('123456');
    expect(wrapper.state().confirmPassword).toEqual('123456');
    expect(instance.validateForm()).toBe(true);
  });

  it('should validate confirmation', () => {
    const confirmEvent = {
      target: {
        id: 'confirmationCode',
        value: "123456"
      }
    }

    // confirmationCode is empty
    expect(wrapper.instance().validateConfirmationForm()).toBe(false);

    // fill confirmationCode
    instance.handleChange(confirmEvent);

    // should be true
    expect(instance.validateConfirmationForm()).toBe(true);
  });

  it('should update the state when submit', () => {
    instance.handleSubmit({ preventDefault: () => { } });
    expect(wrapper.state().newUser).toBe(null);
  });

  test('should show confirmation dialog', () => {
    wrapper.setState({
      newUser: {
        email: 'test@gmail.com',
        password: '123456'
      }
    });
    expect(wrapper.find('HelpBlock').dive().text()).toEqual('Please check your email for the code.')
  });
});