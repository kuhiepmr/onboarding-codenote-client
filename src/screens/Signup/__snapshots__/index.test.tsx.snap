// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp screen should render success 1`] = `
<div
  className="Signup"
>
  <form
    onSubmit={[Function]}
  >
    <FormGroup
      bsClass="form-group"
      bsSize="large"
      controlId="email"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Email
      </ControlLabel>
      <FormControl
        autoFocus={true}
        bsClass="form-control"
        componentClass="input"
        onChange={[Function]}
        type="email"
        value=""
      />
    </FormGroup>
    <FormGroup
      bsClass="form-group"
      bsSize="large"
      controlId="password"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Password
      </ControlLabel>
      <FormControl
        bsClass="form-control"
        componentClass="input"
        onChange={[Function]}
        type="password"
        value=""
      />
    </FormGroup>
    <FormGroup
      bsClass="form-group"
      bsSize="large"
      controlId="confirmPassword"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Confirm Password
      </ControlLabel>
      <FormControl
        bsClass="form-control"
        componentClass="input"
        onChange={[Function]}
        type="password"
        value=""
      />
    </FormGroup>
    <LoaderButton
      block={true}
      bsSize="large"
      disabled={true}
      isLoading={false}
      loadingText="Signing upâ€¦"
      text="Signup"
      type="submit"
    />
  </form>
</div>
`;
